### PR TITLE
Add lambda content

### PR DESCRIPTION
**Pull request by danieltprice**
Deploy preview:
https://websitemain62807-dpriceawslambda2.gatsbyjs.io/docs/guides/aws-lambda/

### CODE REVIEW

File: content/docs/guides/aws-lambda.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+     serverless
+    ```
+
+4. Install the `node-postgres` package, which you will use to connect to the database.
+
+    ```bash
+    npm install pg
+    ```
+
+5. Create a new file named `user.js` and add the following code:
```
Comment (danieltprice) [inline content/docs/guides/aws-lambda.md:75]: Where is user.js created? In the aws-node-project directory that is created by runnign the **serverless** command?

File: content/docs/guides/aws-lambda.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    functions:
+     getAllUsers:
+       handler: users.getAllUsers
+       events:
+         - httpApi:
+             path: /users
+             method: get
+    ```
+
+7. Deploy your function using the following command:
```
Comment (danieltprice) [inline content/docs/guides/aws-lambda.md:127]: The serverless deploy failed for me. Do I require a Serverless Framework account? I got this message:
Warning: Invalid configuration encountered
  at 'org': must match pattern "^[a-z0-9]{5,39}$"
Learn more about configuration validation here: http://slss.io/configuration-validation
Deploying nodejs to stage dev (eu-west-1)
✖ Stack nodejs-dev failed to deploy (0s)
Environment: linux, node 16.17.1, framework 3.26.0, plugin 6.2.2, SDK 4.3.2
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
You are not currently logged in. To log in, use: $ serverless login

File: content/docs/guides/aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+The following steps describe how to create the Lambda function using the [Serverless Framework](https://www.serverless.com/).
+
+1. Install the Serverless Framework by running the following command:
+
+    ```bash
+    npm install -g serverless
+    ```
+
+2. Create a `my-lambda` project and change to that directory.
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:56]:
```suggestion
2. Create a `my-lambda` project directory and navigate into it
```

File: content/docs/guides/aws-lambda.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+       body: JSON.stringify({
+         data: rows,
+       }),
+     };
+    };
+    ```
+
+    The code above exports the function `getAllUsers`, which retrieves all rows from the `users` table and returns them as a `JSON` object in the `HTTP` response body.
+
+    The function uses the `pg` library to connect to the Neon database using the `Client` class and the database connection URL that is stored in the `DATABASE_URL` environment variable. The function calls the connect method on the `Client` instance to establish a connection to the database, and uses the query method to execute a `SELECT` statement that retrieves all rows from the `users` table.
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:97]:
```suggestion
    This function uses the `pg` library to connect to the Neon database. First, you create a new `Client` instance and pass the database connection string which is available in the `DATABASE_URL` environment variable. you then call `connect()` to establish a connection to the database. Finally, you are using the `query()` method to execute a `SELECT` statement that retrieves all rows from the `users` table.
```

File: content/docs/guides/aws-lambda.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    ```js
+    'use strict';
+
+    const { Client } = require('pg');
+
+    module.exports.getAllUsers = async () => {
+     var client = new Client(process.env.DATABASE_URL);
+     client.connect();
+     var { rows } = await client.query('SELECT * from users);
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:85]:
```suggestion
     var { rows } = await client.query('SELECT * from users');
```

File: content/docs/guides/aws-lambda.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+This guide describes how to set up a Neon database and connect to it from an AWS Lambda function using Node.js as the runtime environment. It covers:
+
+- Creating a Lambda function using the [Serverless Framework](https://www.serverless.com/), which is a serverless application lifecycle management framework.
+- Connecting your Lambda function to a Neon database.
+- Deploying the Lambda function to AWS.
+
+## Prerequisites
+
+- A Neon account. If you do not have one, see [Sign up](https://neon.tech/docs/get-started-with-neon/signing-up/) for instructions.
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:16]:
```suggestion
- A Neon account. If you do not have one, see [Sign up](https://neon.tech/docs/get-started-with-neon/signing-up/) for instructions.
- A Neon project. If you do not have one, see [Set up a Project](https://websitemain62807-dpriceupdatebranchapidocs.gatsbyjs.io/docs/get-started-with-neon/setting-up-a-project/)
```

File: content/docs/guides/aws-lambda.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+- An AWS account. You can create a free AWS account at [AWS Free Tier](https://aws.amazon.com/free/).
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:17]:
```suggestion
- An AWS account. You can create a free AWS account at [AWS Free Tier](https://aws.amazon.com/free/). You will also need to [create an IAM User and Access Key](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) to programmatically interact with your AWS account.

```

File: content/docs/guides/aws-lambda.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    ```
+
+2. Create a `my-lambda` project and change to that directory.
+
+    ```bash
+    mkdir my-lambda
+    cd my-lambda
+    ```
+
+3. Run the **serverless** command to create a serverless project. When prompted, select the **AWS - Node.js - Starter** project and provide your AWS account credentials, which include your AWS Access Key Id and AWS Secret Access Key. The process creates an `aws-node-project` directory.
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:63]: I thin here we should show the terminal output when running this command

File: content/docs/guides/aws-lambda.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+    ```bash
+    npm install -g serverless
+    ```
+
+2. Create a `my-lambda` project and change to that directory.
+
+    ```bash
+    mkdir my-lambda
+    cd my-lambda
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:60]:
```suggestion
    mkdir neon-lambda
    cd neon-lambda
```

Just a suggestion

File: content/docs/guides/aws-lambda.md (hunk 1 above)
Comment (m-abdelwahab) [reply content/docs/guides/aws-lambda.md:75]: I think so. Definitely a bit confusing on whether I should install `pg` in the project folder or in the folder created for the lambda function

Comment (danieltprice) [reply content/docs/guides/aws-lambda.md]: Thank you. Fixed.

File: content/docs/guides/aws-lambda.md (hunk 2 above)
Comment (danieltprice) [reply content/docs/guides/aws-lambda.md]: Issue addressed.

File: content/docs/guides/aws-lambda.md (hunk 3 above)
Comment (danieltprice) [reply content/docs/guides/aws-lambda.md:67]: Fixed.

File: content/docs/guides/aws-lambda.md (hunk 6 above)
Comment (danieltprice) [reply content/docs/guides/aws-lambda.md:16]: Added the steps.

File: content/docs/guides/aws-lambda.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+7. Deploy your function using the following command:
+
+    ```bash
+    serverless deploy
+    ```
+
+    The `serverless deploy` command generates an API endpoint using [API Gateway](https://www.serverless.com/framework/docs/providers/aws/events/http-api).
+  
+8. If you make API calls to the Lambda function from your app, you will likely need to configure Cross-Origin Resource Sharing (CORS).  Visit the AWS documentation for information about [how to enable CORS in API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-cors.html).
```
[the remark this answers is not in the document]
Comment (danieltprice) [reply content/docs/guides/aws-lambda.md]: Add a CORS section at the bottom

File: content/docs/guides/aws-lambda.md (hunk 8 above)
Comment (danieltprice) [reply content/docs/guides/aws-lambda.md]: Added!

File: content/docs/guides/aws-lambda.md [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+
+    The query method returns a `Promise` that resolves to an object containing the rows retrieved by the `SELECT` statement, which the function parses to retrieve the `rows` property. Finally, the function returns an `HTTP` response with a status code of 200 and a body that contains a `JSON` object with a single `data` property, which is set to the value of the rows variable.
+
+6. Add the `DATABASE_URL` environment variable and the function definition to the `serverless.yml` file, which is located in your `aws-node-project` directory.
+
+    You can copy the connection string from **Connection Details** widget the Neon Console. Add the `DATABASE_URL` under `environment`, and add `sslmode=require` to the end of the connection string to enable SSL. The `sslmode=require` option tells PostgreSQL to use SSL encryption and verify the server's certificate.
```
Comment (m-abdelwahab) [inline content/docs/guides/aws-lambda.md:172]: Not sure if we should include how to load environment variables from a `.env` file 🤔

https://www.serverless.com/framework/docs/environment-variables. I found this guide in Serverless framework's docs but I haven't tried it

Comment (danieltprice) [reply content/docs/guides/aws-lambda.md:176]: Add a note about using a .env file for environment variables.

**Review by m-abdelwahab — APPROVED** [review]

**Pull request merged**